WIP - add java.time.Duration support for jdocs java files #24646

ralphlaude opened the pull request:
I had java.time.Duration support succesfully for most of the classes in the pull request.
I had java.time.Duration support in this class for the method forMax (`akka.actor.FSM.State#forMax`) and i don't know if it is fine.
The test `akka.stream.scaladsl.UnfoldResourceSourceSpec.scala` the method `"use dedicated blocking-io-dispatcher by default"` is broken and i would like to know how to fix it.

ralphlaude committed:
add java.time.Duration support for jdocs java files #24646

ralphlaude committed:
add java.time.Duration support for jdocs java files #24646

ortex committed:
fix graph stage logging docs

htmldoug committed:
Fix MultiByteArrayIterator.copyToArray silently drops bytes. #16303 (#25141)

I just got burned by #16303.

Answering https://github.com/akka/akka/issues/16303#issuecomment-62957914, it looks like `copyToArray[B >: Byte](Array[B])` can't call `getBytes(Array[Byte], Int, Int)` due to the type constraint (defined by `GenTraversableOnce`).

Instead, I'm going with the simple `isEmpty` solution proposed in the description of #16303 by @kpatrick-kixeye.

raboof committed:
Update scala-java8-compat (#25142)

* Update scala-java8-compat

* Update scala-java8-compat only for scala 2.13

As it looks like 0.8.0 is not bincompat with 0.9.0

ortex committed:
using unidoc for graph stage logging docs

max8github committed:
=doc wording improvement in stream-graphs.md subjunctive (#25145)

raboof committed:
Merge pull request #25139 from ortex/fix-graph-stage-logging-docs

fix graph stage logging docs

chbatey committed:
Typed persistent event adapters/wrappers (#25050)

Typed persistent event adapters/wrappers

richardimaoka committed:
Persistence: replace initialState with emptyState (#25129)

raboof commented:
@ralphlaude hmm, Jenkins seems fine with the 'use dedicated blocking-io-dispatcher by default' test - does it fail for you on master as well? If so perhaps add a separate ticket about that?

raboof commented on akka-docs/src/test/java/jdocs/actor/DependencyInjectionDocTest.java:
```diff
@@ -50,8 +48,9 @@ public static void beforeClass() {
-  public static void afterClass() throws Exception {
-    Await.ready(system.terminate(), Duration.create("5 seconds"));
+  public static void afterClass() {
+    //Await.ready(system.terminate(), Duration.create("5 seconds"));
```
Seems good, remove the commented-out line?

ralphlaude replied:
Done

raboof commented on akka-docs/src/test/java/jdocs/actor/InitializationDocTest.java:
```diff
@@ -29,8 +28,9 @@ public static void beforeClass() {
-  public static void afterClass() throws Exception {
-    Await.ready(system.terminate(), Duration.create("5 seconds"));
+  public static void afterClass() {
+    //Await.ready(system.terminate(), Duration.create("5 seconds"));
```
(idem)

ralphlaude replied:
Done

raboof commented on akka-docs/src/test/java/jdocs/stream/StreamTestKitDocTest.java:
```diff
@@ -102,7 +99,7 @@ public void pipeToTestProbe() throws Exception {
       .runWith(Sink.head(), mat);
     akka.pattern.PatternsCS.pipe(future, system.dispatcher()).to(probe.ref());
-    probe.expectMsg(FiniteDuration.create(3, TimeUnit.SECONDS),
+    probe.expectMsg(scala.concurrent.duration.FiniteDuration.create(3, TimeUnit.SECONDS),
```
I think we have an `expectMsg` that takes a Java duration nowadays, right?

ralphlaude replied:
Done

raboof reviewed (commented):
There might be some overlap with #25060 here, but that's probably fine.

Looking pretty good, couple of comments!

2m committed:
Decrease the uninitialized connection counter as late as possible #25148

This fixes (or decreases the chances significately for) a
situation where a stream of incoming connections is closed
and the incoming connection does not have a chance to startup
properly.

ralphlaude committed:
add java.time.Duration support to all jdocs java file, in the persistence module and the distributed-data module #24646

ralphlaude committed:
add java.time.Duration support to all jdocs java file, in the persistence module and the distributed-data module #24646

ralphlaude commented:
I add one new commit in this pull request. Please check

patriknw committed:
limit negative acknowledgement logging, #25143

patriknw committed:
Merge pull request #25154 from akka/wip-25143-neg-ack-log-patriknw

limit negative acknowledgement logging, #25143

richardimaoka committed:
doc Replace the word stage by operator in stream doc (#25138)

tuleism committed:
Add return type for FlowMonitorState.finished #24885

jiminhsieh committed:
java.time.Duration APIs in javadsl.TestKit #24646

johanandren commented:
I just merged the other PR ( #25060 )  to master so now you will have to rebase (and see what calls was already covered by that)

patriknw committed:
log discarded messages in read-only endpoint #25127

htmldoug committed:
Remove unnecesary drop() from ByteString1.writeToBuffer(ByteBuffer) (#25151)

I found a strange call to `drop()` in `ByteString1.writeToBuffer(ByteBuffer)`. `drop()` instantiates a new `ByteString1` which is immediately discarded to GC. Appears unnecessary.

Removing it makes the `ByteString1` implementation match the one `ByteString1C`.

chbatey committed:
ClusterDeathWatchSpec: assert cluster status bewtween tests (#25126)

Test failed due to node4 still seeing node3 in a subsequent test
but assumed it had been removed.

Fixes #25065

chbatey committed:
Routing docs: Fix typo and clarify group behavior (#25162)

ralphlaude committed:
resolve merge conflicts by adding java.time.Duration support for javadsl #24646

ralphlaude commented:
I rebase the master, Please check

johanandren commented:
Looks like something went wrong with that rebase, the diff looks like it is all changes to master since this branch was created or something.

raboof commented:
@ralphlaude a relatively easy way might be to create a new branch off master, and then `git cherry-pick` only those commits that you want to be part of the PR?

ralphlaude commented:
i will try to find a way.

ralphlaude commented:
Hi guys, the new version of this pull request is here (https://github.com/akka/akka/pull/25206).
Please take a look.
Thanks.

johanandren commented:
Closing this then, as there is a new incarnation
